File: stack.py
class Stack:
    def __init__(self):
        self.storage = {}
        self.size = 0

    def push(self, val):
        self.storage[self.size] = val
        self.size += 1
        return self.size

    def pop(self):
        if self.size:
            result = self.storage[self.size - 1]
            del self.storage[self.size - 1]
            self.size -= 1

            return result
        else: return None

s = Stack()
s.push(1)
s.push(2)
s.push(3)
s.push(4)
s.push(5)

File: stack.js
class Node {
	constructor(val) {
		this.val = val;
		this.next = null;
	}
}

class Stack {
  constructor() {
		this.first = null;
		this.last = null;
		this.size = 0;
	}

	push(val) {
		const node = new Node(val);

		if (!this.size) {
			this.last = node;
		}
		this.node.next = this.first;
		this.first = node;
		this.size++;
		return this.size;
	}

	pop() {
		if (this.size) {
			const first = this.first.val;

			if (this.size === 1) {
				this.first = null;
				this.last = null;
			} else {
				this.first = this.first.next;
			}

			this.size--;
			return first;
		} else return null;
	}
}